packages/legend-graph/src/models/protocols/pure/v1/V1_PureGraphManager.ts

```diff
@@ -2016,6 +2036,237 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
     );
   }
 
+  async indexEntitiesWithDependencyIntoGraph<T extends V1_PackageableElement>(
+    graph: PureModel,
+    _entities: Entity[],
+    dependencyEntities: Map<string, Entity[]>,
+    filteredElementClazzes: GenericClazz<T>[],
+    // TODO we don't need this once protocol type is properly modularized
+    filteredProtocolTypes: string[],
+  ): Promise<V1_GraphBuilderInput[]> {
+    // remove unnecessary entities
+    Array.from(dependencyEntities.entries()).forEach(
```

Comment:
To keep this method generic, I think the best way is not to filter entities here but in the caller and so we could eliminte `filteredElementClazzes` and `filteredProtocolTypes`